src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2302,6 +2302,56 @@ void processContextElement
 
   auto_ptr<DBClientCursor> cursor;
 
+  // This block is to avoids that several entities with the same ID gets updated at the same time, which is
+  // not allowed in NGSIv2. The if is needed, as multiple update has been allowed in NGSIv1 (maybe without
+  // thinking too much about it, but NGSIv1 behaviour now will break backward compatibility)
+  if (apiVersion == "v2")
```

Comment:
I don't like too much using a second operations (`count()`) but I'm not sure the best way of doing. There is a currently open question about this in SOF (http://stackoverflow.com/questions/21832407/mongodb-logging-write-read-queries)

However, I think that the PR can be progressed without getting an answer to that. There are some other cases of count+query in the code and we could fix them all in the same shoot (in this case, I'd add this as technical debt).